core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -716,4 +717,9 @@ default boolean isMaterializedView(Session session, QualifiedObjectName viewName
      * Returns a table handle for the specified table name with a specified version
      */
     Optional<TableHandle> getTableHandle(Session session, QualifiedObjectName tableName, Optional<TableVersion> startVersion, Optional<TableVersion> endVersion);
+
+    /**
+        Returns maximal number of tasks that should be in writing to specific connector.
+     */
+    OptionalInt getMaxWriterTasks(Session session, String catalogName);
```

Comment:
Instead of per catalog, how about we have it per table in a catalog?

Reply:
it could be. But, anyway then you need two methods:
One that takes `tableHandle` and anohter one that takes `tableProperties` (for not-existing tables).
Can you imagine for now any case when it is a table specific?